pokedex/src/Components/App/App.jsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import Header from "../Header/Header";
+import Pokesprites from "../Pokesprites/Pokesprites";
+import Pokeinfo from "../Pokeinfo/Pokeinfo";
+import Aboutme from "../Aboutme/Aboutme";
+import { useState, useEffect } from "react";
+import "../App/App.css";
+import "../style.css";
+import Footer from "../Footer/Footer";
+import { Routes, Route } from "react-router-dom";
+import pokemonData from "../pokemonapi.json";
+import { useNavigate } from "react-router-dom";
+
+const App = () => {
+  const [loading, setLoading] = useState(true);
+  const [url, setUrl] = useState("https://pokeapi.co/api/v2/pokemon/");
+  const [activePokemon, setActivePokemon] = useState();
+  const [pokeInfo, setPokeInfo] = useState({});
+  const [searchTerm, setSearchTerm] = useState("");
+
+  console.log(searchTerm);
+
+  const [pokemonList, setPokemonList] = useState(pokemonData.results);
+  const filteredPokemonList = pokemonList.filter((pokemon) =>
+    pokemon.name.includes(searchTerm)
+  );
+
+  const pokeData = Object.values(pokeInfo).filter((pokemon) => {
+    const bool = pokemon.name.includes(searchTerm);
+    return bool;
+  });
+
+  const navigate = useNavigate();
+
+  const pokeFun = async () => {
+    setLoading(true);
+    const res = await fetch(url);
+    const data = await res.json();
+    fetchPokemonDetails(data.results);
+    setLoading(false);
+  };
+
+  async function getPokemonDetails(url) {
+    try {
+      const res = await fetch(url);
+      if (!res.ok) {
+        throw new Error("Could not get pokemon");
+      }
+      const pokemon = await res.json();
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special function `checkResponse` that will check any response from the server if it’s ok or not.


```js
function checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the function rather than the call. 

----------------------------------------------------------------

**COULD BE IMPROVED**

You can make a special function for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.